src/dc_imex.rs

```diff
@@ -537,33 +540,43 @@ pub unsafe fn dc_normalize_setup_code(
 
 pub unsafe fn dc_job_do_DC_JOB_IMEX_IMAP(context: &Context, job: *mut dc_job_t) {
     let mut current_block: u64;
-    let mut success: libc::c_int = 0i32;
-    let mut ongoing_allocated_here: libc::c_int = 0i32;
+    let mut success: libc::c_int = 0;
+    let mut ongoing_allocated_here: libc::c_int = 0;
     let what: libc::c_int;
-    let mut param1: *mut libc::c_char = 0 as *mut libc::c_char;
-    let mut param2: *mut libc::c_char = 0 as *mut libc::c_char;
+    let mut param1 = 0 as *mut libc::c_char;
+    let mut param2 = 0 as *mut libc::c_char;
+
     if !(0 == dc_alloc_ongoing(context)) {
-        ongoing_allocated_here = 1i32;
-        what = dc_param_get_int((*job).param, DC_PARAM_CMD as i32, 0);
-        param1 = dc_param_get(
-            (*job).param,
-            DC_PARAM_CMD_ARG as i32,
-            0 as *const libc::c_char,
+        ongoing_allocated_here = 1;
+        what = (*job)
+            .param
+            .as_ref()
+            .and_then(|p| p.get_int(Param::Cmd))
+            .unwrap_or_default();
+        param1 = to_cstring(
+            (*job)
+                .param
+                .as_ref()
+                .and_then(|p| p.get(Param::Arg))
+                .unwrap_or_default(),
         );
-        param2 = dc_param_get(
-            (*job).param,
-            DC_PARAM_CMD_ARG2 as i32,
-            0 as *const libc::c_char,
+        param2 = to_cstring(
+            (*job)
+                .param
+                .as_ref()
+                .and_then(|p| p.get(Param::Arg2))
+                .unwrap_or_default(),
         );
-        if param1.is_null() {
+
+        if strlen(param1) == 0 {
             error!(context, 0, "No Import/export dir/file given.",);
         } else {
             info!(context, 0, "Import/export process started.",);
-            context.call_cb(Event::IMEX_PROGRESS, 10i32 as uintptr_t, 0i32 as uintptr_t);
+            context.call_cb(Event::IMEX_PROGRESS, 10 as uintptr_t, 0 as uintptr_t);
```

Comment:
Not a massive fan of stripping all the `i32` bits.  It kind of makes it look like it's been converted while they very likely where some constant originally which should be put back in place.  Leaving them with the `i32` bit makes it obvious they haven't been looked at yet.  (the success/false return values things are fine as i don't think they have constants)